rfcs/0017-long-fork-consensus.md

```diff
@@ -0,0 +1,123 @@
+## Summary
+[summary]: #summary
+
+We currently do not support long forks. This means that to join the network you need a recent checkpoint, in other words to have always been online or to trust another party that has always been online.
+
+This adds support for a client to join the network at any point of time, regardless of if its been online recently. It adds this safely under a [50% + epsilon] security assumption. It also includes a mechanism to restore a chain to strength if this assumption was to be temporarily violated.
+
+This mechanism also does not require checkpointing at any time horizon, compared to alternatives.
+
+## Motivation
+
+[motivation]: #motivation
+
+Its not realistic for clients to have always been online, and it is a poor trust model to require clients to find a trusted third party (high friction and not always feasible).
+
+With this feature, clients will be able to join the network from any device at any time.
+
+## Detailed design
+
+[detailed-design]: #detailed-design
+
+### Part 1: min-window
+
+This version of the implementation is correct given that [50% + epsilon] of stake has always been participating and honest. 
+
+* Add a new field to `protocol_state` called `min_window`, initally set to `8k`
+* Add a new field to `protocol_state` called `current_window`, initially set to 0
+* `let window_diff = FLOOR(current_protocol_state.slot_number/(8k)) - FLOOR(previous_protocol_state.slot_number/(8k))`
```

Comment:
Is this the local slot number or the global slot number? Currently, the slot number tracked within the protocol state is just the local slot number, though the global slot number is simple to derive (epoch_size * epoch_number + local_slot_number).

Reply:
global, I've since fixed